appservice/src/main/java/nu/yona/server/analysis/rest/ActivityController.java

```diff
@@ -0,0 +1,271 @@
+			PagedResourcesAssembler<DayActivityDTO> pagedResourcesAssembler)
+	{
+		return CryptoSession.execute(password, () -> userService.canAccessPrivateData(userID),
+				() -> new ResponseEntity<>(pagedResourcesAssembler.toResource(activityService.getDayActivity(userID, pageable),
+						new DayActivityResourceAssembler(userID)), HttpStatus.OK));
+	}
+
+	@RequestMapping(value = "/weeks/{week}", method = RequestMethod.GET)
+	@ResponseBody
+	public HttpEntity<WeekActivityResource> getWeekActivity(@RequestHeader(value = PASSWORD_HEADER) Optional<String> password,
+			@PathVariable UUID userID, @PathVariable(value = "week") String weekStr, @RequestParam(value = "goal") UUID goalID)
+	{
+		LocalDate date = WeekActivityDTO.parseDate(weekStr);
+		return CryptoSession.execute(password, () -> userService.canAccessPrivateData(userID), () -> new ResponseEntity<>(
+				new WeekActivityResourceAssembler(userID).toResource(activityService.getWeekActivity(userID, date, goalID)),
+				HttpStatus.OK));
+	}
+
+	@RequestMapping(value = "/days/{date}", method = RequestMethod.GET)
+	@ResponseBody
+	public HttpEntity<DayActivityResource> getDayActivity(@RequestHeader(value = PASSWORD_HEADER) Optional<String> password,
+			@PathVariable UUID userID, @PathVariable(value = "date") String dateStr, @RequestParam(value = "goal") UUID goalID)
+	{
+		LocalDate date = DayActivityDTO.parseDate(dateStr);
+		return CryptoSession.execute(password, () -> userService.canAccessPrivateData(userID),
+				() -> new ResponseEntity<>(
+						new DayActivityResourceAssembler(userID).toResource(activityService.getDayActivity(userID, date, goalID)),
+						HttpStatus.OK));
+	}
+
+	public static ControllerLinkBuilder getDayActivityOverviewsLinkBuilder(UUID userID)
+	{
+		ActivityController methodOn = methodOn(ActivityController.class);
+		return linkTo(methodOn.getDayActivityOverviews(null, userID, null, null));
+	}
+
+	public static ControllerLinkBuilder getWeekActivityOverviewsLinkBuilder(UUID userID)
+	{
+		ActivityController methodOn = methodOn(ActivityController.class);
+		return linkTo(methodOn.getWeekActivityOverviews(null, userID, null, null));
+	}
+
+	static ControllerLinkBuilder getDayActivityLinkBuilder(UUID userID, String dateStr, UUID goalID)
+	{
+		ActivityController methodOn = methodOn(ActivityController.class);
+		return linkTo(methodOn.getDayActivity(null, userID, dateStr, goalID));
+	}
+
+	static ControllerLinkBuilder getWeekActivityLinkBuilder(UUID userID, String weekStr, UUID goalID)
+	{
+		ActivityController methodOn = methodOn(ActivityController.class);
+		return linkTo(methodOn.getWeekActivity(null, userID, weekStr, goalID));
+	}
+
+	static class WeekActivityResource extends Resource<WeekActivityDTO>
+	{
+		public WeekActivityResource(WeekActivityDTO weekActivity)
+		{
+			super(weekActivity);
+		}
+	}
+
+	static class WeekActivityOverviewResource extends Resource<WeekActivityOverviewDTO>
```

Comment:
This class corresponds with a collection of week activity overviews, right? I wonder why that is required. We don't have a class for this in the ``MessageControler``, right?